File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.js
/**
 * @param {string} s
 * @return {boolean}
 */
var validPalindrome = function (s) {
    let first = 0;
    let end = s.length - 1
    while (first < end) {
        if (s[first] != s[end]) {
            return check(s, first + 1, end) || check(s, first, end - 1)
        }
        first++
        end--
    }
    return true
}
var check = function (s, left, right) {
    while (left < right) {
        if (s[left] != s[right]) return false
        left++
        right--
    }
    return true
}

File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        # civic
        # civibc

        left = 0
        right = len(s) - 1
        count = 0


        while left < right:
            if s[left] != s[right]:
                full_string = s
                remove_l = s[:left] + s[left+1:]
                remove_r = s[:right] + s[right+1:]
                if remove_l == remove_l[::-1] or remove_r == remove_r[::-1]:
                    return True
                else:
                    return False
            left+=1
            right-=1

        return True
